src/hotspot/share/runtime/arguments.cpp

```diff
@@ -4114,6 +4114,13 @@ jint Arguments::apply_ergo() {
     return code;
   }
 
+  if (_mode == _int) {
+    if (SegmentedCodeCache) {
+      warning("SegmentedCodeCache has no meaningful effect with -Xint");
+      FLAG_SET_DEFAULT(SegmentedCodeCache, false);
+    }
+  }
```

Comment:
Isn't this better placed in Arguments::set_mode_flags where the other compiler related flags get turned off?

Reply:
`SegmentedCodeCache` should be set in `compiler/compilerDefinitions.cpp`:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/compiler/compilerDefinitions.cpp#L294

Reply:
I replied on email but I don't see it and this is better here.  I should really change this to be:

if (FLAG_IS_CMDLINE(SegmentedCodeCache)) {
}
Since this code is checking for someone setting it on the command line with -Xint.  The code in compilerDefinitions.cpp sets the flag ergonomically if the code cache is big and if tiered compilation is on.  I believe this is after the code in arguments.cpp.  So this code won't reset it to true, if I'm following it right.

Reply:
I agree that the warning should be issued if flag is set on command line together with `-Xint`.
In this case you need add this check to `CompilerConfig::check_args_consistency()` where different compiler flags are set to false when we run with Interpreter:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/compiler/compilerDefinitions.cpp#L493

Reply:
Thanks Coleen, seems the use of `set_mode_flags` is somewhat flawed as you can change any of the flags that it tries to force on/off simply by having them later on the command-line.

You could do as @vnkozlov suggests and move this check into `compilerDefinitions.cpp` as `CompilerConfig::ergo_initialize()` is called just a few lines above this.

Reply:
I moved the check but TieredStopAtLevel=0 still allocates code heaps for nmethods, so the check only checks for the -Xint in arguments. I don't know if this makes sense or not in the code heap allocation code, but the codecache/cli tests test for sizes of code heaps pass with this change.